packages/gatsby/src/query/file-parser.js

```diff
@@ -58,13 +58,31 @@ function followVariableDeclarations(binding) {
 }
 
 function isUseStaticQuery(path) {
-  return (
-    (path.node.callee.type === `MemberExpression` &&
-      path.node.callee.property.name === `useStaticQuery` &&
-      path.get(`callee`).get(`object`).referencesImport(`gatsby`)) ||
-    (path.node.callee.name === `useStaticQuery` &&
-      path.get(`callee`).referencesImport(`gatsby`))
-  )
+  const callee = path.node.callee
+  if (callee.type === `MemberExpression`) {
+    const property = callee.property
+    if (property.name === `useStaticQuery`) {
+      return path.get(`callee`).get(`object`).referencesImport(`gatsby`, ``)
+    }
+    return false
+  }
```

Comment:
just for the record - this code path won't cover commonjs case without destructuring - i.e.:
```
const gatsby = require(`gatsby`)
[...]
gatsby.useStaticQuery(...)
```
But I think this is not a blocker for this PR